app/src/main/java/com/github/se/signify/model/challenge/ChallengeRepositoryFireStore.kt

```diff
@@ -14,34 +14,94 @@ class ChallengeRepositoryFireStore(private val db: FirebaseFirestore) : Challeng
       onSuccess: () -> Unit,
       onFailure: (Exception) -> Unit
   ) {
+    // Generate initial words for each round
+    val roundWords =
+        listOf("apple", "banana", "cherry") // Replace with dynamic word generation if needed
+
```

Comment:
Consider dynamic word generation or sourcing from a database instead of hardcoding round words (apple, banana, cherry). This makes the solution more scalable and adaptable for future requirements.

Reply:
As you can see with my comment, it was already taken into account and i was also going to do it for my next task, but nice catch it's necessary to do it!